cmake/CMakeLists.txt

```diff
@@ -280,35 +283,19 @@ if(BUILD_MPI)
   # We use a non-standard procedure to cross-compile with MPI on Windows
   if((CMAKE_SYSTEM_NAME STREQUAL "Windows") AND CMAKE_CROSSCOMPILING)
     include(MPI4WIN)
-    target_link_libraries(lammps PUBLIC MPI::MPI_CXX)
   else()
     find_package(MPI REQUIRED)
-    target_link_libraries(lammps PUBLIC MPI::MPI_CXX)
     option(LAMMPS_LONGLONG_TO_LONG "Workaround if your system or MPI version does not recognize 'long long' data types" OFF)
     if(LAMMPS_LONGLONG_TO_LONG)
       target_compile_definitions(lammps PRIVATE -DLAMMPS_LONGLONG_TO_LONG)
     endif()
   endif()
+  target_link_libraries(lammps PUBLIC MPI::MPI_CXX)
 else()
-  file(GLOB MPI_SOURCES ${LAMMPS_SOURCE_DIR}/STUBS/mpi.cpp)
-  add_library(mpi_stubs STATIC ${MPI_SOURCES})
-  set_target_properties(mpi_stubs PROPERTIES OUTPUT_NAME lammps_mpi_stubs${LAMMPS_MACHINE})
-  target_include_directories(mpi_stubs PUBLIC $<BUILD_INTERFACE:${LAMMPS_SOURCE_DIR}/STUBS>)
-  if(BUILD_SHARED_LIBS)
-    target_link_libraries(lammps PRIVATE mpi_stubs)
-    if(MSVC)
-      target_link_libraries(lmp PRIVATE mpi_stubs)
-      target_include_directories(lmp INTERFACE $<BUILD_INTERFACE:${LAMMPS_SOURCE_DIR}/STUBS>)
-      target_compile_definitions(lmp INTERFACE $<INSTALL_INTERFACE:LAMMPS_LIB_NO_MPI>)
-    endif()
-    target_include_directories(lammps INTERFACE $<BUILD_INTERFACE:${LAMMPS_SOURCE_DIR}/STUBS>)
-    target_compile_definitions(lammps INTERFACE $<INSTALL_INTERFACE:LAMMPS_LIB_NO_MPI>)
-  else()
-    target_include_directories(lammps INTERFACE $<BUILD_INTERFACE:${LAMMPS_SOURCE_DIR}/STUBS>)
-    target_compile_definitions(lammps INTERFACE $<INSTALL_INTERFACE:LAMMPS_LIB_NO_MPI>)
-    target_link_libraries(lammps PUBLIC mpi_stubs)
-  endif()
-  add_library(MPI::MPI_CXX ALIAS mpi_stubs)
+  target_sources(lammps PRIVATE ${LAMMPS_SOURCE_DIR}/STUBS/mpi.cpp)
+  add_library(mpi_stubs INTERFACE)
+  target_include_directories(mpi_stubs INTERFACE $<BUILD_INTERFACE:${LAMMPS_SOURCE_DIR}/STUBS>)
+  target_link_libraries(lammps PUBLIC mpi_stubs)
```

Comment:
What happened to the alias?

Reply:
rolled that change back to minimize complications while working on other portability issues. will check it out again, now that everything else seems to be sorted out.

Reply:
@junghans here is what I learned about this issue:
The alias as it was done before would not work because we already created the `MPI::MPI_CXX` imported target when using `find_package(MPI QUIET)` to initialize the `BUILD_MPI` option setting. CMake lets be create the alias, but it will use the imported target instead. Since there is no way to remove/delete/disable/rename an imported target, the only viable solution seems to be to create the same alias target for **both**. So I created a target `MPI::ANY_CXX` to represent either the real MPI or the stub MPI library and when I use *this* alias, it will correctly use the stub definitions as expected. At least as far as I could find out so far.

Reply:
sadly, now ubuntu 18.04LTS (or rather CMake 3.10) is rearing its ugly head. While this works for serial compilation, the MPI compilation fails because CMake 3.10 does not let me create an alias to an imported target. For later CMake it will work, after the imported target is promoted to global scope, but that option does not exist. So back to square one and the old CMake code until ubuntu 18 is obsolete and we can require a later CMake version.

Reply:
Too bad.

Could we fix that by using:
```
set_target_properties(MPI::MPI_CXX PROPERTIES IMPORTED_GLOBAL TRUE)
```
on CMake-3.10 or below ONLY?

A another option would be to do the alias only for 3.12 and above, and add that to the if block for `target_link_libraries()` that we have to carry around either way and then it is at least clear in the future what to do. Because it seems like every 6 months we are asking yourselves again why we didn't use an `ALIAS` target ;-)

Reply:
No. The property exists since 3.11 only.
I think having those ifs for cmake versions are worse than the current situation.

Reply:
There is not much more left to streamline. This version is still a significant improvement since the settings are carried with the LAMMPS library and we don't need to look for and link to the stubs library, which was causing the problems that got me started.

Reply:
I would at least leave a comment recommending the future use of an ALIAS target.